ml-workflow-demo-kfp-katib-mlflow/README.md

```diff
@@ -0,0 +1,472 @@
+                ]
+            }
+        }
+    }
+    
+    # Create the KFP task for the TFJob.
+    tfjob_launcher_op = components.load_component_from_url(
+"https://raw.githubusercontent.com/kubeflow/pipelines/master/components/kubeflow/launcher/component.yaml")
+    op = tfjob_launcher_op(
+        name=tfjob_name,
+        namespace=tfjob_namespace,
+        chief_spec=json.dumps(tfjob_chief_spec),
+        worker_spec=json.dumps(tfjob_worker_spec),
+        tfjob_timeout_minutes=60,
+        delete_finished_tfjob=False)
+    return op
+```
+
+10. Define a complete pipeline that consists of all steps created earlier. Note that the name of the pipeline must be uqinue. If there was previously defined pipeline with the same name and within the same namespace either change the name of current pipeline or delete the older pipeline from the namespace.
+
+
+```python
+name = "demo-pipeline"
+namespace = "admin"
+s3_bucket = "demo-dataset"
+key = "data.csv"
+dataset_url = "https://www.openml.org/data/download/53995/KDDCup09_churn.arff"
+
+@dsl.pipeline(
+    name = "ML Workflow in Kubeflow",
+    description = "Demo pipeline"
+)
+def demo_pipeline(name=name, namespace=namespace):
+
+    # Step 1: Download dataset.
+    ingest_data_task = ingest_data_op(url=dataset_url)
+
+    # Step 2: Clean up the dataset and store it in S3 bucket.
+    clean_data_op = components.create_component_from_func(
+            clean_arff_data,
+            "clean_data.yaml",
+            packages_to_install=["pandas==1.2.4", "scipy==1.7.0", "boto3"],
+        )
+    clean_data_task = (clean_data_op(
+        s3_bucket,
+        key,
+        ingest_data_task.outputs['data']
+    ).apply(use_k8s_secret(
+        secret_name='mlpipeline-minio-artifact',
+        k8s_secret_key_to_env={
+            'accesskey': 'AWS_ACCESS_KEY_ID',
+            'secretkey': 'AWS_SECRET_ACCESS_KEY',
+        })))
+
+    with dsl.Condition(clean_data_task.output == "Done"):
+        # Step 3: Run hyperparameter tuning with Katib.
+        katib_op = create_katib_experiment_op(name, namespace, s3_bucket, key)
+
+        # Step 4: Convert Katib results produced by hyperparameter tuning to model.
+        convert_katib_results_op = components.func_to_container_op(convert_katib_results)
+        best_katib_model_op = convert_katib_results_op(katib_op.output)
+
+        # Step 5: Run training with TFJob. Model will be stored into ML Flow model registry
+        # (done inside container image).
+        tfjob_op = create_tfjob_op(name, namespace, best_katib_model_op.output, s3_bucket, key)
+```
+
+11. Execute pipeline.
+
+
+```python
+kfp_client = Client()
+run_id = kfp_client.create_run_from_pipeline_func(
+        demo_pipeline,
+        namespace=namespace,
+        arguments={},
+    ).run_id
+print(f"Run ID: {run_id}")
+```
+
+12. Observe run details by selecting **Run details** link.
+
+![Run](./images/ML-Workflow-RunDetails.png)
+
+![Pipeline](./images/ML-Workflow-Pipeline.png)
+
```

Comment:
At this point, we can also tell them to view the Katib experiment in the Katib UI (`Experiments (AutoML)` in the dashbaord).  As soon as the katib launcher starts, their experiemnt's progress will be visible there.  We can add a image here too with an example, that way they're not just hoping that katib finishes soon